lib/decorators/placeholders.js

```diff
@@ -167,8 +168,6 @@ export function hasPlaceholder(uri, path, storeOverride = store) {
     isCollapsibleList = isComponentList && !!_.get(schema, `${componentListProp}.collapse`),
     isComponentProp = schema && !!schema[componentProp];
 
-  // if there isn't a placeholder at all, return negative quickly
-  // since we're ususally checking a bunch of fields/groups
   if (!placeholder) {
     return false;
   }
```

Comment:
no explanation needed for early returns because they're the best. It's so much nicer reserving your unindented code for your main logical branch and your indented code for early returns, error handling, etc.